ui/src/onboarding/containers/LoginPageContents.tsx

```diff
@@ -0,0 +1,383 @@
+            gradient={Gradients.GarageBand}
+          >
+            {notificationText}
+          </Notification>
+        </Panel>
+      </form>
+    )
+  }
+
+  private get validFieldValues(): {
+    isValid: boolean
+    errors: {[fieldName: string]: string}
+  } {
+    const {
+      activeTab,
+      firstName,
+      lastName,
+      email,
+      password,
+      confirmPassword,
+    } = this.state
+
+    const firstNameError = !firstName && 'This field is required'
+    const lastNameError = !lastName && 'This field is required'
+    const emailError = !email && 'This field is required'
+    const passwordError = !password && 'This field is required'
+    const confirmPasswordError =
+      confirmPassword === password
+        ? !confirmPassword && 'This field is required'
+        : 'Passwords must match'
+
+    const errors: ErrorObject = {
+      emailError,
+      passwordError,
+    }
+    if (activeTab === 'signup') {
+      errors.firstNameError = firstNameError
+      errors.lastNameError = lastNameError
+      errors.confirmPasswordError = confirmPasswordError
+    }
+
+    const isValid = !Object.values(errors).some(error => !!error)
+
+    return {isValid, errors}
+  }
+
+  private handleSubmit = (e: FormEvent) => {
+    const {isValid, errors} = this.validFieldValues
+    const {
+      email,
+      password,
+      firstName: given_name,
+      lastName: family_name,
+      activeTab,
+    } = this.state
+
+    e.preventDefault()
+
+    if (!isValid) {
+      this.setState(errors)
+      return
+    }
+
+    this.setState({buttonStatus: ComponentStatus.Loading})
+
+    if (activeTab === 'login') {
+      auth0.login(
+        {
+          realm: Auth0Connection.Authentication,
+          email,
+          password,
+        },
+        err => {
+          if (err) {
+            this.setState({buttonStatus: ComponentStatus.Default})
+            return this.displayErrorMessage(errors, err)
+          }
+        }
+      )
+      return
+    }
+
+    auth0.signup(
+      {
+        connection: Auth0Connection.Authentication,
+        email,
+        password,
+        family_name,
+        given_name,
+      },
+      err => {
+        if (err) {
+          this.displayErrorMessage(errors, err)
+          this.setState({buttonStatus: ComponentStatus.Default})
+          return
+        }
+        // log the user into Quartz
+        auth0.login(
+          {
+            realm: Auth0Connection.Authentication,
+            email,
+            password,
+          },
+          error => {
+            if (error) {
+              this.setState({buttonStatus: ComponentStatus.Default})
+              return this.displayErrorMessage(errors, error)
+            }
+          }
+        )
+      }
+    )
+  }
+
+  private displayErrorMessage = (errors, auth0Err) => {
+    if (auth0Err.code.includes('error in email')) {
+      this.setState({
+        ...errors,
+        emailError: 'Please enter a valid email address',
```

Comment:
Since the auth0Err is provided to us and we aren't performing any actionable email parsing (purposefully), it doesn't seem like we need to elaborate on an email address's validity